### PR TITLE
Handle session present error case

### PR DESCRIPTION
**Pull request by leegeth**
*Description of changes:*
Handle an error case when a session present flag is received from
broker when a clean session is requested.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/mqtt/src/mqtt.c [hunk 1]
```diff
@@ -997,6 +1000,19 @@ static MQTTStatus_t receiveConnack( const MQTTContext_t * pContext,
...
 
+    /* If a clean session is requested, a session present should not be set by
+     * broker. */
+    if( status == MQTTSuccess )
+    {
+        if( ( cleanSession == true ) && ( *pSessionPresent == true ) )
+        {
+            LogError( ( "Received CONNACK with session present from broker."
+                        " This should not happen as a clean session is requested"
+                        " in CONNECT." ) );
```
Comment (aggarw13) [inline libraries/standard/mqtt/src/mqtt.c:1011]: Probably should be a warning instead of an error as this represents an issue in the broker,
Also, recommend making the message concise as:
```suggestion
            LogWarn( ( "Unexpected session present flag in CONNACK response from broker."
                        " CONNECT request with clean session was made with broker." ) );
```

File: libraries/standard/mqtt/utest/mqtt_utest.c [hunk 2]
```diff
@@ -815,6 +827,20 @@ void test_MQTT_Connect_happy_path()
...
+    /* Request to establish a session if present and session present is received
+     * from broker. */
+    mqttContext.connectStatus = MQTTNotConnected;
+    connectInfo.cleanSession = false;
+    sessionPresent = true;
+    MQTT_GetIncomingPacketTypeAndLength_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTT_GetIncomingPacketTypeAndLength_ReturnThruPtr_pIncomingPacket( &incomingPacket );
+    MQTT_DeserializeAck_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTT_DeserializeAck_ReturnThruPtr_pSessionPresent( &sessionPresent );
+    status = MQTT_Connect( &mqttContext, &connectInfo, NULL, timeout, &sessionPresentResult );
```
Comment (muneebahmed10) [inline libraries/standard/mqtt/utest/mqtt_utest.c:840]: The other tests pass `sessionPresent` to MQTT_Connect. I'd swap these two around and possibly change the name of the return pointer to `sessionPresentExpected` or something to be consistent with the other tests

File: libraries/standard/mqtt/utest/mqtt_utest.c [hunk 3]
```diff
@@ -691,6 +692,17 @@ void test_MQTT_Connect_receiveConnack( void )
     MQTT_DeserializeAck_ExpectAnyArgsAndReturn( MQTTBadResponse );
     status = MQTT_Connect( &mqttContext, &connectInfo, NULL, timeout, &sessionPresent );
     TEST_ASSERT_EQUAL_INT( MQTTBadResponse, status );
+
+    /* Session present is received when a clean session is requested. */
```
Comment (aggarw13) [inline libraries/standard/mqtt/utest/mqtt_utest.c:696]: (Nit): Can be clearer with "Test case when broker sends session present flag in response to a clean session connection request. "

File: libraries/standard/mqtt/utest/mqtt_utest.c [hunk 4]
```diff
@@ -691,6 +692,17 @@ void test_MQTT_Connect_receiveConnack( void )
...
+    mqttContext.transportInterface.recv = transportRecvSuccess;
+    connectInfo.cleanSession = true;
+    sessionPresent = true;
+    MQTT_GetIncomingPacketTypeAndLength_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTT_GetIncomingPacketTypeAndLength_ReturnThruPtr_pIncomingPacket( &incomingPacket );
+    MQTT_DeserializeAck_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTT_DeserializeAck_ReturnThruPtr_pSessionPresent( &sessionPresent );
```
Comment (muneebahmed10) [inline libraries/standard/mqtt/utest/mqtt_utest.c:703]: I'd add a variable so the pointer that gets returned by MQTT_DeserializeAck isn't identical to the one passed to MQTT_Connect

File: libraries/standard/mqtt/src/mqtt.c (hunk 1 above)
Comment (leegeth) [reply libraries/standard/mqtt/src/mqtt.c:1011]: I think I still would prefer the error as it is a condition in which a connection cannot be continued with. I like the sentences you suggested :)

File: libraries/standard/mqtt/utest/mqtt_utest.c (hunk 2 above)
Comment (leegeth) [reply libraries/standard/mqtt/utest/mqtt_utest.c:840]: ok.

File: libraries/standard/mqtt/utest/mqtt_utest.c [hunk 5]
```diff
@@ -999,6 +1013,7 @@ void test_MQTT_Connect_happy_path()
     MQTT_GetIncomingPacketTypeAndLength_ReturnThruPtr_pIncomingPacket( &incomingPacket );
     MQTT_DeserializeAck_ExpectAnyArgsAndReturn( MQTTSuccess );
     MQTT_DeserializeAck_ReturnThruPtr_pSessionPresent( &sessionPresentExpected );
+    MQTT_PubrelToResend_ExpectAnyArgsAndReturn( MQTT_PACKET_TYPE_INVALID );
```
Comment (aggarw13) [inline libraries/standard/mqtt/utest/mqtt_utest.c:1016]: Why was this needed to be added?

Comment (leegeth) [reply libraries/standard/mqtt/utest/mqtt_utest.c:1016]: Rebase brought the resend logic, so a call to function MQTT_PubrelToResend will be made if a session is present.
